plugins/modules/email_alert.py

```diff
@@ -0,0 +1,263 @@
+
+# from typing import List, Tuple, Union, Dict, Any
+
+
+def create_email_alert(module: AnsibleModule, rest_client: RestClient):
+    before = EmailAlert.get_state(rest_client)
+    for item in before:
+        if item.get("email_address") == module.params["email"]:
+            return False, before, dict(before=before, after=before)
+    EmailAlert.create(
+        rest_client=rest_client,
+        payload=dict(emailAddress=module.params["email"]),
+        check_mode=module.check_mode,
+    )
+    after = EmailAlert.get_state(rest_client)
+    return (
+        after != before,
+        after,
+        dict(before=before, after=after),
+    )  # changed, records, diff
+
+
+def update_email_alert(module: AnsibleModule, rest_client: RestClient):
+    before = EmailAlert.get_state(
+        rest_client
+    )  # get the records of emailAlert endpoint before update
+
+    # Get record of old emailAlert by email
+    old_email = EmailAlert.get_by_email(
+        dict(email_address=module.params["email"]), rest_client
+    )
+
+    if not old_email:
+        old_email = EmailAlert.get_by_email(
+            dict(email_address=module.params["email_new"]), rest_client
+        )
+        if not old_email:
+            raise errors.ScaleComputingError(
+                "Email Alert: Can't update a nonexistent email."
+            )
+
+    # Return if there are no changes
+    if (
+        module.params["email_new"] == old_email.to_ansible().get("email_address")
+        or module.params["email_new"] == module.params["email"]
+    ):
+        return False, before, dict(before=before, after=before)
+
+    # Otherwise, update with new email address
+    old_email.update(
+        rest_client=rest_client,
+        payload=dict(emailAddress=module.params["email_new"]),
+        check_mode=module.check_mode,
+    )
+    after = EmailAlert.get_state(
+        rest_client
+    )  # get the records od emailAlert endpoint after update
+    return (
+        after != before,
+        after,
+        dict(before=before, after=after),
+    )  # changed, records, diff
+
+
+def delete_email_alert(module: AnsibleModule, rest_client: RestClient):
+    before = EmailAlert.get_state(rest_client)
+    delete_email = EmailAlert.get_by_email(
+        dict(email_address=module.params["email"]), rest_client
+    )
+    if not delete_email:
+        module.warn("Email Alert: The email you're trying to remove, doesn't exist.")
+        return False, before, dict(before=before, after=before)
+    delete_email.delete(rest_client, module.check_mode)
+
+    after = EmailAlert.get_state(rest_client)
+    return (
+        after != before,
+        after,
+        dict(before=before, after=after),
+    )  # changed, records, diff
+
+
+def send_test(module: AnsibleModule, rest_client: RestClient):
+    before = EmailAlert.get_state(rest_client)
+
+    if module.params["email_new"] is not None:
+        module.warn("Email Alert: parameter 'email_new' is not needed.")
```

Comment:
Can we ask ansible to prevent this - required_if or similar?

Reply:
I was looking for something like ``not_required_if`` or ``hidden_if`` because, even when we use ``state=present`` the parameter ``email_new`` is not required. Unless we make the parameter ``email_new`` as required when ``state=present``, but when we don't want to update the current email, we just set ``email_new`` to ``""``?